### PR TITLE
upgrade to go1.23

### PR DESCRIPTION
**Pull request by cpanato**
- upgrade to go1.23

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/sigstore/fulcio
 
-go 1.22.6
+go 1.23
```
Comment (haydentherapper) [inline go.mod:3]: Is this fine without the specific version or toolchain? I would prefer this but not sure if go.mod requires otherwise.

**Comment by cpanato** [comment]
@haydentherapper would you mind to take another look?

File: Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.22.6@sha256:2bd56f00ff47baf33e64eae7996b65846c7cb5e0a46e0a882ef179fd89654afa AS builder
+FROM golang:1.23.0@sha256:613a108a4a4b1dfb6923305db791a19d088f77632317cfc3446825c54fb862cd AS builder
```
Comment (haydentherapper) [inline Dockerfile:16]: bump to 1.23.1?

Comment (cpanato) [reply Dockerfile]: done PTAL

**Review by haydentherapper — APPROVED** [review]
thanks!

**Pull request merged**